### PR TITLE
examples: Google Analytics with TypeScript

### PR DESCRIPTION
**Pull request by tarunama**
## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Errors have helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have helpful link attached, see `contributing.md`

## Documentation / Examples

- [x] Make sure the linting passes by running `pnpm lint`
- [x] The examples guidelines are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples)

### CODE REVIEW

File: examples/with-google-analytics-typescript/lib/gtags.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const GA_TRACKING_ID = process.env.NEXT_PUBLIC_GA_ID ?? ''
+
+// measure PV
+const pageview = (url: string): void => {
+  if (GA_TRACKING_ID === '') return
+
+  window.gtag('config', GA_TRACKING_ID, {
+    page_path: url,
+  })
```
Comment (SukkaW) [inline examples/with-google-analytics-typescript/lib/gtags.ts:9]: IMHO it is not the correct way of doing this.

Check [`gtags.js` documentation](https://developers.google.com/analytics/devguides/collection/gtagjs/pages#manual_pageviews) about how to manually send page view event on SPA.

File: examples/with-google-analytics-typescript/pages/_app.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+import '../styles/globals.css'
+import { GA_TRACKING_ID, pageview } from '../lib/gtags'
+
+function MyApp({ Component, pageProps }: AppProps) {
+  // google analyticsの設定
+  const router = useRouter()
+
+  useEffect(() => {
+    if (GA_TRACKING_ID === undefined) return
```
Comment (SukkaW) [inline examples/with-google-analytics-typescript/pages/_app.tsx:14]:
```
const GA_TRACKING_ID = process.env.NEXT_PUBLIC_GA_ID ?? ''
```

So `GA_TRACKING_ID` will never be `undefined`.

File: examples/with-google-analytics-typescript/pages/_app.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import type { AppProps } from 'next/app'
+import { useRouter } from 'next/router'
+import { useEffect } from 'react'
+import Script from 'next/script'
+
+import '../styles/globals.css'
+import { GA_TRACKING_ID, pageview } from '../lib/gtags'
+
+function MyApp({ Component, pageProps }: AppProps) {
+  // google analyticsの設定
```
Comment (SukkaW) [inline examples/with-google-analytics-typescript/pages/_app.tsx:10]:
```suggestion
```

**Review by SukkaW — CHANGES_REQUESTED** [review]

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Please have a look at our contributing guidelines and adjust your PR accordingly for a review: https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples

Eg.: Note there is already a [`with-google-analytics`](https://github.com/vercel/next.js/tree/canary/examples/with-google-analytics) example, let's just convert that to TS.

I also think the `pnpm-lock.yaml` file has been changed by accident, could you revert that?

**Pull request closed without merging**